src/adapter_test_helpers/sumo_basic_test_helper.erl

```diff
@@ -176,6 +177,24 @@ check_proper_dates(Config) ->
   Date = Module:birthdate(Person),
   ok.
 
+-spec persist_using_changeset(config()) -> ok.
+persist_using_changeset(Config) ->
+  {_, Name} = lists:keyfind(name, 1, Config),
+  Module = sumo_config:get_prop_value(Name, module),
+  Schema = Module:sumo_schema(),
+
+  [] = sumo:find_by(Name, [{name, <<"John">>}]),
+  [P1] = sumo:find_by(Name, [{name, <<"A">>}]),
+  Doc1 = sumo:new_doc(people, P1),
```

Comment:
And exposing `sumo:new_doc` might lead to that… I would prefer having `sumo_internal:new_doc` and implementing `sumo_changeset:cast/4` in this way…

```erlang
cast(SchemaName, UserDoc, Params, Allowed) ->
  cast(sumo_internal:new_doc(SchemaName, UserDoc), Params, Allowed).
```

In short, I would try as much as I can to keep our internal formats strictly internal and not exposed to the regular sumo user.